tsl/src/chunk.c

```diff
@@ -893,6 +893,14 @@ chunk_merge_chunks(PG_FUNCTION_ARGS)
 					 errmsg("merging compressed chunks is not yet supported"),
 					 errhint("Decompress the chunks before merging.")));
 
+		if (ts_chunk_is_frozen(chunk))
+			ereport(ERROR,
+					(errcode(ERRCODE_FEATURE_NOT_SUPPORTED),
+					 errmsg("cannot merge frozen chunk \"%s.%s\"",
```

Comment:
Users don't explicitly freeze chunk. Perhaps a more user friendly message is: 
```suggestion
					 errmsg("cannot merge frozen chunk . Chunk scheduled for tiering\"%s.%s\"",
```

and the corresponding hint could be: call untier_chunk to unfreeze the chunk

Reply:
Probably better to use as a detailed message, but I agree with @gayyappan that it is not very clear if the user does not explicitly freeze the chunk. Maybe rather use "`cannot merge chunk \"%s.%s\"scheduled for tiering`". I know the freeze flag can be used more generally, but this is where we are now.

Reply:
@gayyappan What happens if someone unfreezes a chunk, i.e., calls `unfreeze_chunk` which is scheduled for tiering?

Reply:
freeze/unfreeze are not user level APIS. These are called internally only by tier_chunk/untier_chunk today.
So if you call unfreeze_chunk directly on a chunk scheduled for tiering - it will leave things in a bad state for tiering. The chunk can still get tiered and/or dropped from Postgres.